### PR TITLE
Improve branding resolver cache clearing logic

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
Previously, cache clearing was implemented by looping (breadth-first) through the child organization hierarchy. If the number of suborganizations at a particular level is high, it will result in a database hit

### CODE REVIEW

File: components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java [hunk 1]
```diff
@@ -215,32 +221,58 @@ public void clearBrandingResolverCacheHierarchy(String currentTenantDomain) thro
...
-            try {
-                String childTenantDomain = organizationManager.resolveTenantDomain(childOrganizationId);
-                if (StringUtils.isNotBlank(childTenantDomain)) {
-                    BrandedOrgCacheEntry valueFromCache =
-                            brandedOrgCache.getValueFromCache(brandedOrgCacheKey, childTenantDomain);
-                    if (valueFromCache != null) {
-                        // If cache exists, clear the cache
-                        brandedOrgCache.clearCacheEntry(brandedOrgCacheKey, childTenantDomain);
+        String cursor = null;
+        int pageSize = 2;
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java:242]: why this is two?

File: components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java [hunk 2]
```diff
@@ -353,32 +385,60 @@ public void clearCustomTextResolverCacheHierarchy(String currentTenantDomain, St
...
 
-            try {
-                String childTenantDomain = organizationManager.resolveTenantDomain(childOrganizationId);
-                TextCustomizedOrgCacheEntry valueFromCache =
-                        textCustomizedOrgCache.getValueFromCache(cacheKey, childTenantDomain);
-                if (valueFromCache != null) {
-                    // If cache exists, clear the cache.
-                    textCustomizedOrgCache.clearCacheEntry(cacheKey, childTenantDomain);
+        String cursor = null;
+        int pageSize = 2;
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java:407]: why page size is two?

Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java]: updated

File: components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java (hunk 1 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java]: updated.

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**